File: fib.jl
using DataFrames
using BenchmarkTools
using CSV

# Fibonacci function - iterative implementation
function fibonacci(n)
    a, b = 0, 1
    for _ in 1:n
        a, b = b, a + b
    end
    return a
end

# Initialize a DataFrame to store the results
df_julia = DataFrame(Run = Int[], Time = Float64[])

# Define the number of runs on each batch
num_runs = 100

# Warm up the JIT compiler
fibonacci(1_000_000)

# Loop over the number of runs
println("Go, Julia!")
for i in 1:num_runs
    run_time = @elapsed fibonacci(1_000_000)
    push!(df_julia, [i run_time])
	println("Run: $(i)")
end

#println(df_julia)
CSV.write("Output.csv", df_julia)

File: fib.py
import pandas as pd
import timeit

#Fibonacci function - iterative implementation
def fibonacci(n):
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a

# Initialize a DataFrame to store the results
df_python = pd.DataFrame(columns=['Run', 'Time'])

# Define the number of runs on each batch
num_runs = 10

# Loop over the number of runs
print("Go Python!")
for i in range(num_runs):
    start_time = timeit.default_timer()
    fibonacci(1_000_000)
    end_time = timeit.default_timer()
    run_time = end_time - start_time
    df_python = df_python.append({'Run': i+1, 'Time': run_time}, ignore_index=True)
    print(f'Run: {i+1}', end='\r')

print(df_python)
df_python.to_csv('Outputs.csv', index=False)